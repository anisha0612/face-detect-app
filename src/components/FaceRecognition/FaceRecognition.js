import React from "react";

const FaceRecognition = ({ imageUrl, boxes }) => {
  return (
    <div className='center ma'>
      <div className='absolute'>
        <img
          id='image'
          alt='pic'
          src={imageUrl}
          width='500px'
          height='auto'
          className='mt3'
        />
        {boxes.map((box) => (
          <div
            key={box.topRow}
            className='flex absolute flex-wrap center pointer shadow-2 ba bw2 b--light-blue '
            style={{
              top: box.topRow,
              left: box.leftCol,
              right: box.rightCol,
              bottom: box.bottomRow,
            }}></div>
        ))}
      </div>
    </div>
  );
};

export default FaceRecognition;
