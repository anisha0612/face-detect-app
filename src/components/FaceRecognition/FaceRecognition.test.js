import React from "react";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";
import FaceRecognition from "./FaceRecognition";

describe("Face Recognition Component ", () => {
  it("creating a snapshot of the face recognition component", () => {
    const mockImgUrl = "http://image.jpg";
    const mockBoxes = [
      { topRow: 23, bottomRow: 45, leftCol: 59, rightCol: 34 },
      { topRow: 40, bottomRow: 75, leftCol: 49, rightCol: 24 },
    ];
    expect(
      create(
        <FaceRecognition imageUrl={mockImgUrl} boxes={mockBoxes} />
      ).toJSON()
    ).toMatchSnapshot();
  });
});
