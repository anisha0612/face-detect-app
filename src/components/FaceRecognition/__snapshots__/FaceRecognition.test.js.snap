// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Face Recognition Component  creating a snapshot of the face recognition component 1`] = `
<div
  className="center ma"
>
  <div
    className="absolute"
  >
    <img
      alt="pic"
      className="mt3"
      height="auto"
      id="image"
      width="500px"
    />
    <div
      className="flex absolute flex-wrap center pointer shadow-2 ba bw2 b--light-blue "
      style={
        Object {
          "bottom": 45,
          "left": 59,
          "right": 34,
          "top": 23,
        }
      }
    />
    <div
      className="flex absolute flex-wrap center pointer shadow-2 ba bw2 b--light-blue "
      style={
        Object {
          "bottom": 75,
          "left": 49,
          "right": 24,
          "top": 40,
        }
      }
    />
  </div>
</div>
`;
